src/node/coinstats.cpp

```diff
@@ -66,9 +88,38 @@ bool GetUTXOStats(CCoinsView* view, CCoinsStats& stats, const std::function<void
         pcursor->Next();
     }
     if (!outputs.empty()) {
-        ApplyStats(stats, ss, prevkey, outputs);
+        ApplyStats(stats, hash_obj, prevkey, outputs);
     }
-    stats.hashSerialized = ss.GetHash();
+
+    FinalizeHash(hash_obj, stats);
+
     stats.nDiskSize = view->EstimateSize();
     return true;
 }
+
+bool GetUTXOStats(CCoinsView *view, CCoinsStats& stats, CoinStatsHashType hash_type, const std::function<void()>& interruption_point)
+{
+    switch(hash_type) {
+        case(CoinStatsHashType::HASH_SERIALIZED): {
+            CHashWriter ss(SER_GETHASH, PROTOCOL_VERSION);
+            return GetUTXOStats(view, stats, ss, interruption_point);
+        }
+        case(CoinStatsHashType::NONE): {
+            return GetUTXOStats(view, stats, nullptr, interruption_point);
+        }
+    }
+    assert(false); // Unreachable code
```

Comment:
Should be identical to the other cases to make it grep-able. See dev-notes example:

```cpp
enum class Tabs {
    INFO,
    CONSOLE,
    GRAPH,
    PEERS
};

int GetInt(Tabs tab)
{
    switch (tab) {
    case Tabs::INFO: return 0;
    case Tabs::CONSOLE: return 1;
    case Tabs::GRAPH: return 2;
    case Tabs::PEERS: return 3;
    } // no default case, so the compiler can warn about missing cases
    assert(false);
}
```
```suggestion
    assert(false);
```

Reply:
Ok, thanks. Fixed.